### ISSUE TITLE
Xournal++ continuously crash using a lot of resource

### ISSUE THREAD

**Issue by Frequel** [xournalpp/xournalpp#1904]
- Ubuntu
 - Gnome-Wayland
 - version of libgtk : 3.24.18
 - Version of Xournal++: 1.0.16-3-g1429879c5a
 - Installation method:snap

Hardware:
 - XPS-13-9343
 - Intel® Core™ i7-5600U CPU @ 2.60GHz × 4 
 - RAM 8GB DDR3L‑RS 1600MHz
 - SSD M.2 256GB

when i am annotating a PDF adding image and text it require a lot of time to load the page and sometime doesn't show the annotation that i just made, it will appear only after a saving that will take around 5-10 minutes. When i export in PDF instead i require at least 15 minutes. Really often i can't do anything of this because Xournal freeze and Ubuntu ask me if i want force the quit or wait and if i wait i will wait for ever..... when is freezed or when is saving i open monitor app and xournal is using the 30% of cpu and 4GB of RAM (i have 8GB but it can't take more because is used from others resource i think). 
EDIT:  as i though, it take more RAM as soon as i kill other programs in the background, now it has taken 4,7GB

Steps to reproduce the behavior (of last crash):
1. Add 3 blank page
2. Add a short text
3. Try to add an image

It will freeze on the image selection and after few seconds ubuntu will ask for you if you want force quit

**Comment by Technius** [comment]
Does this occur when you're working on a new document (not with a background PDF)?

**Comment by tpapp** [comment]
I am experiencing something similar (90%+ CPU) latest master (22a0dd89a0), using Ubuntu 20.04, for both blank documents and drawing on a PDF. Another way to reproduce is the highlighter tool, it produces brief spikes in CPU usage.

**Comment by rolandlo** [comment]
Is this still an issue with the latest master?

**Comment by rolandlo** [comment]
Closing due to inactivity.

**Comment by r0hit2005** [comment]
Hey, I recently installed the xournalpp on ubuntu 22.04 via the official repository. The app seems to crash when I load a pdf for annotating, or when I scroll down an already loaded pdf. Sometimes, it crashes while I am writing something. I usually ask it to wait if it's not loading but sometimes the app crashes completely and I need to load the pdf again and even lose a few of the latest annotations.

The app does not seem to use any unreasonable amount of system resources either. 

Versions:
Xournal++ 1.1.1
└──libgtk: 3.24.33

Specs: 
Acer Nitro AN515-42
AMD Ryzen 5 2500U
8GB RAM
Vega 8 Graphics / Radeon RX 560X dedicated
Ubuntu 22.04 
Wacom One Tablet

**Comment by Technius** [comment]
Please open another issue with your crash logs.
